### ISSUE TITLE
Wrap DMFieldEvaluate and expose it to users

### ISSUE THREAD

**Issue by julesghub** [underworldcode/underworld3-archive#235]
DMFieldEvaluate looks like it works with DMPlex and retrieves values, 1st and 2nd derivatives for a field.
https://petsc.org/release/manualpages/DM/DMFieldEvaluate/

Yes to having access to this feature for visualisation, model checking and debugging.
Wrapping it involves some extra cython code and a way to pythonically caputure the output data.

@bknight1

**Comment by knepley** [comment]
Yes, this was Toby's idea. It is a _different object_ named DMField, which is designed to be evaluatable over the mesh. We currently use this for interaction withcoordinates.

**Comment by julesghub** [comment]
Excellent @knepley ! Is it likely to be wrapped anytime soon in petsc4py? That would be super.
